src/test/java/be/syntra/devshop/DevshopBack/services/SearchServiceTest.java

```diff
@@ -0,0 +1,218 @@
+package be.syntra.devshop.DevshopBack.services;
+
+import be.syntra.devshop.DevshopBack.entities.Product;
+import be.syntra.devshop.DevshopBack.models.SearchModel;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import java.math.BigDecimal;
+
+import static be.syntra.devshop.DevshopBack.testutilities.ProductUtils.createDummyPageable;
+import static be.syntra.devshop.DevshopBack.testutilities.ProductUtils.createDummyProductPage;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class SearchServiceTest {
+
+    @Mock
+    ProductService productService;
+
+    @InjectMocks
+    SearchServiceImpl searchService;
+
+    @Test
+    void findAllNonArchivedBySearchTermAndPriceBetween() {
+        // given
+        SearchModel dummySearchModel = new SearchModel();
```

Comment:
removing boilerplatecode by extracting the SearchModel? Perhaps convert it to a builder()-method, where you give the parameters? (the last 4 never change)

Reply:
add all fields to the dummyDefaultSearchModel with dummyData, so that when others call the method, they'll know what to expect. 
If someone else adds parameters you forgot, they'll break your tests without directly knowing why

Reply:
used an existing method to make a dummy searchModel and removed the new one